frontend/controller/actions/group.js

```diff
@@ -220,18 +219,23 @@ export default (sbp('sbp/selectors/register', {
         const chatRoomIds = Object.keys(rootState[params.contractID].chatRooms)
           .filter(cId => rootState[params.contractID].chatRooms[cId].users.includes(me))
 
+        /**
+         * flag READY_TO_JOIN_CHATROOM is not necessary to sync actually
+         * But just this is only for checking if syncing chatrooms or not
+         * Especially inside addMentioning in model/contracts/chatroom.js
+         */
+        sbp('okTurtles.data/set', 'READY_TO_JOIN_CHATROOM', true)
         await sbp('chelonia/contract/sync', chatRoomIds)
+        sbp('okTurtles.data/set', 'READY_TO_JOIN_CHATROOM', false)
 
         sbp('state/vuex/commit', 'setCurrentChatRoomId', {
           groupId: params.contractID,
           chatRoomId: rootState[params.contractID].generalChatRoomId
         })
       }
       sbp('okTurtles.data/set', 'JOINING_GROUP', false)
-      sbp('okTurtles.data/set', 'READY_TO_JOIN_CHATROOM', false)
     } catch (e) {
       sbp('okTurtles.data/set', 'JOINING_GROUP', false)
-      sbp('okTurtles.data/set', 'READY_TO_JOIN_CHATROOM', false)
```

Comment:
I don't understand this. Can you please explain more in detail?

Reply:
Yeah, in the new file, on line 228 you have:

```js
        await sbp('chelonia/contract/sync', chatRoomIds)
```

If there's an exception there, then `sbp('okTurtles.data/set', 'READY_TO_JOIN_CHATROOM', true)` will not get reset because this line was removed.

Reply:
Got it. Thanks